[Compute] Dedicate host cmdlets

hyonholee opened the pull request:
Design review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/260

<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Please add a brief description of the changes made in this PR -->

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

cormacpayne commented on src/Compute/Compute/Generated/Models/PSGalleryImageVersion.cs:
```diff
@@ -40,7 +40,8 @@ public string ResourceGroupName
 
-        public GalleryImageVersionPublishingProfile PublishingProfile { get; set; }
+        // This is commented out for hiding the breaking change caused by the client library update.
+        //public GalleryImageVersionPublishingProfile PublishingProfile { get; set; }
```
@hyonholee what's the breaking change that we're trying to hide here? Since we're commenting this property out, we're introducing a breaking change. Since this is a preview module, we can just make the breaking change introduced by the SDK.

hyonholee replied:
GalleryImageVersionPublishingProfile used to have "source" property in 2019-03-01 API version.

https://github.com/Azure/azure-rest-api-specs/blob/d8e61eccaaa0f9ddeb95dfed0a6dcebe06dbbc3b/specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/gallery.json#L1438

Now, galley team moved the property to GalleryImageVersionStorageProfile in 2019-07-01 version.

https://github.com/Azure/azure-rest-api-specs/blob/d8e61eccaaa0f9ddeb95dfed0a6dcebe06dbbc3b/specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/gallery.json#L1154

cormacpayne commented on src/Compute/Compute/help/Get-AzHost.md:
````diff
@@ -0,0 +1,159 @@
+
+### Example 1
+```
+PS C:\> Get-AzHost -ResourceGroupName $resourceGroupName -HostGroupName $hostGroupName -Name $hostName
````
@hyonholee would you mind adding the corresponding output for the commands being run in the examples for the new cmdlets? This will let users see what is returned when running the commands.

cormacpayne commented on src/Compute/Compute/Generated/DedicatedHost/DedicatedHostDeleteMethod.cs:
```diff
@@ -0,0 +1,135 @@
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        public SwitchParameter Force { get; set; }
```
@hyonholee do we need the `-Force` parameter here? We only need to include this parameter if there is a reason to have an additional prompt for the user. If this command is deleting a resource _and_ its sub-resources, then I think it's OK to have the `-Force` since we want to prompt the user to confirm they want to delete the additional sub-resources.

cormacpayne commented on src/Compute/Compute/Generated/DedicatedHost/DedicatedHostCreateOrUpdateMethod.cs:
```diff
@@ -0,0 +1,132 @@
+        [Parameter(
+            Position = 3,
+            Mandatory = true)]
+        public string Location { get; set; }
```
@hyonholee would you mind adding the `LocationCompleter` to this parameter?

cormacpayne commented on src/Compute/Compute/Generated/DedicatedHostGroup/DedicatedHostGroupCreateOrUpdateMethod.cs:
```diff
@@ -0,0 +1,105 @@
+        [Parameter(
+            Position = 2,
+            Mandatory = true)]
+        public string Location { get; set; }
```
@hyonholee same comment about adding `LocationCompleter`

cormacpayne commented on src/Compute/Compute/Generated/DedicatedHostGroup/DedicatedHostGroupDeleteMethod.cs:
```diff
@@ -0,0 +1,124 @@
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        public SwitchParameter Force { get; set; }
```
@hyonholee same comment about whether or not we need `-Force`

hyonholee commented:
@cormacpayne   Added LocationCompleter for New-AzHost and New-AzHostGroup. Removed Force for Remove-AzHost and Remove-AzHostGroup. Added output examples for Get-AzHost and Get-AzHostGroup. Updated chagnelog.   Thanks.
